src/framework-ondata.js

```diff
@@ -52,17 +52,19 @@ module.exports = function(payload, socketIndex) {
          if ( ! _.isFunction( this.nodeInstance.wrapper.fRunUpdater ) ) { 
 
              // Don't have a wrapper fRunUpdater 
+             profiler.stopEvent(eventStart);
              throw Error( 'No wrapper fRunUpdater function found!  Cannot run updater.' );
 
          } else { 
-
              // TODO: revisit this logic later, per rdf-pipeline/noflo-rdf-pipeline#35
              // save data used to see if we have a different state after running updater
              clearStateData( vni.errorState ); // clear last error state
 
              // Save wrapper to make it accessible from the Promise
              var wrapper = this.nodeInstance.wrapper;
-             var updaterStartTime = Date.now();
+
+             profiler.stopEvent(eventStart); // done with general event processing 
+             var updateStart = profiler.startUpdate(); 
```

Comment:
It could be, but I'm not sure that is better.  We want to ensure that all elapsed time is covered - no gaps for any reason ->  Either the promise set up time should count against the event set up or against the updater.    If I move it down inside the Promise (moving that time onto event set up) that feels wrong since the event set up is complete at this point.  Also the code becomes less readable IMHO.  Let's talk about this.